server/api/api/endpoints/projects.py

```diff
@@ -399,6 +415,17 @@ async def get_project_summary(
         server.api.api.deps.authenticate_request
     ),
 ):
+    if (
+        mlrun.mlconf.httpdb.clusterization.role
+        != mlrun.common.schemas.ClusterizationRole.chief
+    ):
+        logger.info(
+            "Requesting project summaries, re-routing to chief",
+            name=name,
+        )
+        chief_client = server.api.utils.clients.chief.Client()
+        return await chief_client.get_project_summary(project=name, request=request)
+
```

Comment:
i wouldnt delegate the responsibility to chief only. why not letting each worker/chief do it same as we have today?
later on, upon separation / shared queue then we can do it in one place and distribute to others

Reply:
Making the workers collect project summaries trades network load for cpu load. IMO since the chief is doing it anyway, the network overhead will be insignificant in comparison to the cpu load on all workers. Also, that means you can make 2 requests to get the project summaries and get 2 different responses because each worker holds something else in cache

Reply:
> Making the workers collect project summaries trades network load for cpu load. IMO since the chief is doing it anyway, the network overhead will be insignificant in comparison to the cpu load on all workers. Also, that means you can make 2 requests to get the project summaries and get 2 different responses because each worker holds something else in cache

I agree the latter is not ideal, but that is what we have now. overall, not optimal at all but fact we need to redirect to chief is another hop/api load on chief we're trying to reduce
if that's a hassle - perhaps simply table of project summaries on DB with single record where  chief updates / workers reads could be ideal